crates/miden-lib/asm/miden/contracts/auth/basic.masm

```diff
@@ -11,47 +12,82 @@ const.AUTH_REQUEST=131087
 # The slot in this component's storage layout where the public key is stored.
 const.PUBLIC_KEY_SLOT=0
 
-#! Authenticate a transaction using the Falcon signature scheme
+#! Authenticate a transaction using the Falcon signature scheme.
+#!
+#! It first increments the nonce of the account, independent of whether the account's state has
+#! changed or not. Then it computes and signs the following message (in memory order):
+#! [ACCOUNT_DELTA_COMMITMENT, INPUT_NOTES_COMMITMENT, OUTPUT_NOTES_COMMITMENT, [0, 0, 0, final_nonce]]
```

Comment:
As mentioned offline, I would also include the reference block number of the transaction into the salt.